### ISSUE TITLE
The problem of calculating the spherical harmonics

### ISSUE THREAD

**Issue by Editi0** [moble/spherical#9]
When I use this program to calculate the spherical harmonics(s=0), the results are obviously problematic (compared to [scipy](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.sph_harm.html)), is there something wrong with my code?

```python
import numpy as np
import matplotlib.pyplot as plt
import spherical
import quaternionic
from scipy.special import sph_harm

ell = 2
m = 2
theta = np.linspace(0, np.pi, 100)

R = quaternionic.array.from_spherical_coordinates(theta, 0)
y1 = [spherical.Wigner(10).sYlm(0, r)[spherical.Wigner(2).Yindex(ell, m)] for r in R]
y2 = np.abs(sph_harm(m, ell, 0, theta))
plt.plot(theta, y1, label = 'spherical')
plt.legend()
plt.plot(theta, y2, label = 'scipy')
plt.legend()
```

![image](https://user-images.githubusercontent.com/39005998/110327535-105c6c00-8055-11eb-99b8-6eeab29b1be0.png)

**Comment by moble** [comment]
Sorry about that; it's my fault.  The problem is with the `Yindex` member, which calls another function with the arguments in the wrong order.  What you're actually seeing is the ell=1 m=0 mode.  I don't actually use that member, and only tested the function that was getting called, so I hadn't run into this bug.  But I've fixed it now and added tests for that behavior.  The corrected version (1.0.5) is now available on PyPI, and should appear on conda-forge later today.

---

While we're at it, here are a few points about code quality.  The code as you've written it will do a lot more computation than it needs to, and is susceptible to bugs when things change.

First, the `Wigner` object should be created as infrequently as possible because it has to do some pre-computation.  Typically, you should just pick the largest `ell` value you'll need, and create it once with that value (unless you're doing huge calculations that could really benefit from various values of `ell_min` and `mp_max`).

Also, it's dangerous to use different `Wigner` objects to compute values and index into those computations.  Specifically, using `10` and `2` here is a bad idea:
```python
spherical.Wigner(10).sYlm(0, r)[spherical.Wigner(2).Yindex(ell, m)]
```
because if you ever decide to change `mp_max` or `ell_min`, you would likely run into errors.  Just pick one value and stick with it.  Again, this won't be a problem if you just create a single `Wigner` object, and use it for both jobs.

Finally, this update simplifies calling the `sYlm` method so that you don't need to loop over `R`; you can just call `spherical.Wigner(10).sYlm(0, R)`.  The various modes will be stored in the final dimension of the output.

Putting these all together, your code would look like
```python
import numpy as np
import matplotlib.pyplot as plt
import spherical
import quaternionic
from scipy.special import sph_harm

ell = 2
m = 2
theta = np.linspace(0, np.pi, 100)

wigner = spherical.Wigner(ell)

R = quaternionic.array.from_spherical_coordinates(theta, 0)
y1 = wigner.sYlm(0, R)[..., wigner.Yindex(ell, m)]
y2 = sph_harm(m, ell, 0, theta)
plt.plot(theta, y1.real, label='spherical')
plt.plot(theta, y2.real, label='scipy')
plt.legend();
```

**Comment by Editi0** [comment]
Thanks for your response and points about my code.
